eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,144 @@
+
+Write-Host "Updating autorest.md files for all the changed swaggers."
+$sdksInfo = @{}
+$headSha = $inputFileContent.headSha
+$repoHttpsUrl = $inputFileContent.repoHttpsUrl
+foreach ($inputFilePath in $inputFilePaths) {
+  foreach ($path in $autorestFilesPath) {
+    $fileContent = Get-Content $path
+    $isUpdatedLines = $false
+    $updatedLines = foreach ($line in $fileContent) {
+      $regexForMatchingShaAndPath = "[\/][0-9a-f]{4,40}[\/]$inputFilePath"
+      if ($line -match $regexForMatchingShaAndPath) {
+        $line -replace "https:\/\/[^`"]*$regexForMatchingShaAndPath", "$repoHttpsUrl/blob/$headSha/$inputFilePath"
+
+        $isUpdatedLines = $true
+        $sdkpath = (get-item $path).Directory.Parent.FullName | Resolve-Path -Relative
+        if (!$sdksInfo.ContainsKey($sdkpath)) {
+          $specReadmePath = $inputFileContent.relatedReadmeMdFiles -match $inputFilePath
+          $sdksInfo.Add($sdkpath, $specReadmePath)
+        }
+      }
+      else {
+        $line
+      }
+    }
+    if ($isUpdatedLines) {
+      $updatedLines | Out-File -FilePath $path
+    }
+  }
+}
+
+Write-Host "Updated autorest.md files for all the changed swaggers. `n"
+
+function Test-PreviousScript() {
+  $result = $null
+  if ($LASTEXITCODE -eq 0) {
+    $result = "succeeded"
+  }
+  else {
+    $result = "failed"
+  }
+  Write-Host "Result: $result"
+  return $result
+}
+
+$packages = @()
+foreach ($sdkPath in $sdksInfo.Keys) {
+  $packageName = Split-Path $sdkPath -Leaf
+  Write-Host "Generating code for " $packageName
+  $srcPath = Join-Path $sdkPath 'src'
+  dotnet msbuild /restore /t:GenerateCode $srcPath
+
+  $artifacts = @()
+  $hasBreakingChange = $null
+  $content = $null
+  $result = Test-PreviousScript
+  if ($result -eq "succeeded") {
+    Write-Host "Successfully generated code for" $packageName "`n"
+    
+    $csprojPath = Get-ChildItem $srcPath -Filter *.csproj -Recurse
+    dotnet pack $csprojPath /p:RunApiCompat=$false
+    $result = Test-PreviousScript
+    if ($result -eq "succeeded") {
+      $artifactsPath = "$RepoRoot/artifacts/packages/Debug/$packageName"
+      $artifacts += Get-ChildItem $artifactsPath -Filter *.nupkg -Recurse | Select-Object -ExpandProperty FullName | Resolve-Path -Relative
+
+      $logFilePath = Join-Path "$srcPath" 'log.txt'
+      if (!(Test-Path $logFilePath)) {
+        New-Item $logFilePath
+      }
+      dotnet build $csprojPath /t:RunApiCompat /p:TargetFramework=netstandard2.0 /flp:v=m`;LogFile=$logFilePath
+      
+      $result = Test-PreviousScript
+      if ($result -eq "succeeded") {
+        $hasBreakingChange = $false
+      }
+      else {
+        $logFile = Get-Content -Path $logFilePath | select-object -skip 2
+        $breakingChanges = $logFile -join ",`n"
+        $content = "Breaking Changes: $breakingChanges"
+        $hasBreakingChange = $true
+        $result = "succeeded"
```

Comment:
I'd stick with `$true` and `$false` rather than inventing your own return codes. It's idiomatic, and easier to maintain. That's what `Test-Path` is returning, for example.

Reply:
The values for $result are fixed and it can be either succeeded or failed in generateOuput.json in[ SDK automation](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4254356&view=logs&j=7f026a02-f755-5ce4-425c-2d5df22c9d55&t=20b436ef-0412-5b55-7996-5294882c57b5&l=271) step. If I set any other value it will fail.